Python/Algorithms/aditya/traversal.py

```diff
@@ -0,0 +1,123 @@
+from collections import deque
+class Traversal:
+    @staticmethod
+    def inorder( root ,l): 
+  
+        if root is not None: 
+            Traversal.inorder(root.left,l) 
+            l.append(root.val )
+            Traversal.inorder(root.right,l) 
+        return l 
+    @staticmethod
+    def postorder(root,l): 
+    
+        if root is not None: 
+            Traversal.postorder(root.left, l) 
+            Traversal.postorder(root.right , l) 
+            l.append(root.val )
+        return l
+    @staticmethod
+    def preorder(root , l): 
+    
+        if root is not None: 
+
+            l.append(root.val )
+            Traversal.preorder(root.left , l) 
+            Traversal.preorder(root.right , l) 
+        return l
+    @staticmethod
+    def levelorder(root , l): 
+        if root is None: 
+            return
+        q = deque([])
+        q.append(root) 
+        while len(q):
+            current = q.popleft()
+            l.append(current.val)  
+            if current.left : 
+                q.append(current.left) 
+            if current.right : 
+                q.append(current.right)
+        return l
```

Comment:
This is same as BFS. replace it with BFS algorithm

Reply:
I dont quite get your meaning , its the same algorithm as bfs , but this one just traverses the
 tree in bfs fashion , should i replace this to search for an element like bfs ?

Reply:
A bfs will also print how it traversed along with the key existence

Reply:
so i have updated the code to return a bool indicating found or not found with also a list of items it traversed before finding the element .